### PR TITLE
Fix OEC report check, add test

### PR DESCRIPTION
**Pull request by bamader**
Fast follows to fix the OEC report query. Updates income determination parsing to not generate enrollment parsing bugs and updates fake data generator to create 'not determined' incomes for foster records. Adds an integration test for a submitted roster to make sure oec-report route works. Solves flashing issue on home page when determining whether to show pre or post submit.

### CODE REVIEW

**Comment by bamader** [comment]
Okay so this PR fixes up the submit functionality and corrects a not disclosed bug that was messing with parsing, but I cannot figure out why nightwatch refuses to click the submit button on the roster page in the e2e test I made and it's 9:30 on a Friday night so I need to  table this for now.

**Review by jlhogan — APPROVED** [review]

**Comment by bamader** [comment]
Not sure what was up the help buttons but it apparently liked the styling we used on the pre-submit home page, so I used links with arrow icons and that now shows up in staging and QA.

**Comment by jlhogan** [comment]
weird! let's open a ticket to investigate in the future why those buttons didnt work but i think completely and totally ok for now!! ty !!

**Comment by RyanHansz** [comment]
@bamader Button links look good! The link to the help page isn't working ATM, it goes to:
https://help.ece-reporter.ctoec.org/

**Comment by bamader** [comment]
That might be from a qa test run where I'd only rerun the integration tests instead of rebuilding/re-redeploying. I just completed a passing run and I'm seeing that it goes to the correct https://ece-reporter.ctoec.org/help on qa. Wanna confirm @RyanHansz ?

**Comment by RyanHansz** [comment]
> That might be from a qa test run where I'd only rerun the integration tests instead of rebuilding/re-redeploying. I just completed a passing run and I'm seeing that it goes to the correct https://ece-reporter.ctoec.org/help on qa. Wanna confirm @RyanHansz ?

To clarify, this link is going to https://ece-reporter.ctoec.org/help instead of https://help.ece-reporter.ctoec.org/

Screen Shot 2021-01-11 at 11 52 52 AM

**Comment by bamader** [comment]
Sorry, my brain is moving slow this morning. #687 says in the AC that the link to use is https://ece-reporter.ctoec.org/help. But that is not the case, and we actually want the link to go to https://help.ece-reporter.ctoec.org/?

**Comment by RyanHansz** [comment]
> Sorry, my brain is moving slow this morning. #687 says in the AC that the link to use is https://ece-reporter.ctoec.org/help. But that is not the case, and we actually want the link to go to https://help.ece-reporter.ctoec.org/?

That's the desired URL for the page itself (with the two buttons). The button link is in my last comment

File: client/src/containers/Help/Help.tsx [hunk 1]
```diff
@@ -45,12 +46,17 @@ const Help: React.FC = () => {
             </div>
             <h2>How-to guides</h2>
             <p>Find your answers to common tasks and questions.</p>
-
-            <Button
-              text="View guides"
-              href="https://ece-reporter.ctoec.org/help"
-              external={true}
-            />
+            <a
```
Comment (JordanGuinn) [inline client/src/containers/Help/Help.tsx:49]: Minor question: Should this link open the Help Guides in a new browser tab?

**Review by JordanGuinn — APPROVED** [review]

File: client/src/containers/Help/Help.tsx (hunk 1 above)
Comment (RyanHansz) [reply client/src/containers/Help/Help.tsx:49]: Sure, that'd be best

**Comment by RyanHansz** [comment]
@bamader this looks good to me once the link is updated

File: client/src/containers/Help/Help.tsx (hunk 1 above)
Comment (bamader) [reply client/src/containers/Help/Help.tsx:49]: Dumb question incoming: is there a way to make a link open in a new tab automatically?

**Pull request merged**